R/functions_netCDF.R

```diff
@@ -456,42 +687,377 @@ create_netCDF_from_array_with_variables <- function(x, locations, grid,
+
+    nc_names <- attributes(nc$var)$names
+    stopifnot(var_names[k] %in% nc_names)
+    var_nc_index <- grep(var_names[k], nc_names)[1]
+
+    nc_var <- ncdf4::ncvar_get(nc, attributes(nc$var)$names[var_nc_index])
+    nc_var_dims <- dim(nc_var) # lon, lat, then #vars time or depth of gridded 
+
+    if(verbose) {
+      message(paste('The dimensionality of variable',var_names[k], 'is',
+                    paste(nc_var_dims, collapse = ', ')))
+    }
+    
+    # Set up chunksizes  ---------------------------------------------------
+    if(has_Z_verticalAxis) {
+      if(has_T_timeAxis) { # if time and depth are both TRUE , the third dimension
+        z_chunksize <- dim(data)[3]
+      } 
+    }
+    
+    # ---------------------------------------------------------------------
+    # NOT GRIDDED ---------------------------------------------------------
+    # ---------------------------------------------------------------------
+    if(isGridded == FALSE) {
+      
+      var_chunksizes <- nc_var_dims[1] # site
+      
+      #  -----------------------------------------------------------------------
+      # add variable values ! --------------------------------------------------
+      # ------------------------------------------------------------------------
+      if(nvars > 1) {
+        
+        vals <- data[ ,k]
+        
+        try(ncdf4::ncvar_put(nc, varid = var_names[k],
+                             vals = vals,
+                             start = 1,
+                             count = var_chunksizes))
+        
+      } else {
+        if(has_T_timeAxis && has_Z_verticalAxis) {
+          
+          for(z in seq(z_chunksize)) { # by Z axis
+            if (verbose) message('Adding depth layer ', z)
+            for (t in seq_len(nn)) { # col by col - always time in this case
+              
+              vals <- data[, t, z]
+              
+              try(ncdf4::ncvar_put(nc, varid = var_names[k],
+                                   vals = vals,
+                                   start = c(1, t, z), #x-y-t-z
+                                   count = c(var_chunksizes, 1, 1)))
+            }
+          }
+        } else { 
+          # write values, col by col -- n values can rep dif. vals, time, or depth
+          for (n in seq_len(nn)) {
+            
+            vals <-  data[, n] # by time chunk or var chunk
+
+            var_start <-  c(1, n)
+            
+            try(ncdf4::ncvar_put(nc, varid = var_names[k],
+                                 vals = vals,
+                                 start = var_start,
+                                 count = c(var_chunksizes, 1)))
+          }
+        }
+      }
+    }
+    
+    #  GRIDDED --------------------------------------------------------------
+    if(isGridded == TRUE) {
+      
+      # more checks based on dim of var ----------------------------------------
+      
+      if(has_T_timeAxis & !has_Z_verticalAxis) stopifnot(nc_var_dims[3] == nn)
```

Comment:
lines 1004-1005: replace `&` with `&&`
--> even better, combine lines 1004 and 1005 `if (isTRUE(xor(has_T_timeAxis, has_Z_verticalAxis)))`

Reply:
Never seen xor. Cool!